db/tables.py

```diff
@@ -408,4 +413,46 @@ def infer_table_column_types(
             table_name,
             column_name,
             engine,
+            conn=conn,
         )
+
+
+class CreateTempTableAs(DDLElement):
+    def __init__(self, name, selectable):
+        self.name = name
+        self.selectable = selectable
+
+
+@compiler.compiles(CreateTempTableAs)
+def compile(element, compiler, **_):
+    return "CREATE TEMPORARY TABLE %s AS (%s)" % (
+        element.name,
+        compiler.sql_compiler.process(element.selectable, literal_binds=True),
+    )
+
+
+def infer_table_column_types(schema, table_name, engine):
+    table = reflect_table(table_name, schema, engine)
+    temp_name = "temp_table"
```

Comment:
I think this should be randomly-chosen on the fly (or random-ish:  Think `temp_table_<uuid>` or something similar).  The reason is that PostgreSQL puts temp tables into a kind of "global" space, and while they're not _accessible_ from other sessions, they block creation of a table with the same name in those session.  This would lead to problems if more than one client is connected, trying to use this function.